src/app/checkout/step-2/existingPaymentMethods/existingPaymentMethods.component.js

```diff
@@ -61,20 +61,21 @@ class ExistingPaymentMethodsController {
     }
   }
 
-  openAddNewPaymentMethodModal(){
-    this.addNewPaymentMethodModal = this.$uibModal.open({
-      component: 'addNewPaymentMethodModal',
+  openPaymentMethodFormModal(){
+    this.paymentMethodFormModal = this.$uibModal.open({
+      component: 'paymentMethodFormModal',
       size: 'large new-payment-method-modal',
       backdrop: 'static', // Disables closing on click
       resolve: {
         submissionError: this.submissionError,
+        mailingAddress: this.mailingAddress,
         onSubmit: () => (params) => {
           params.stayOnStep = true;
           this.onSubmit(params);
         }
       }
     });
-    this.addNewPaymentMethodModal.result.then(null, () => {
+    this.paymentMethodFormModal.result.then(null, () => {
```

Comment:
Instead on then(null, Function), you can use catch(Function) to drop the need for the initial null. Internally I think catch calls then with a null. Catch might be a little cleaner.